diff -apurN a/source/settings/languagemanager.cpp b/source/settings/languagemanager.cpp
--- a/source/settings/languagemanager.cpp	2013-01-23 00:04:36.435199303 +0100
+++ b/source/settings/languagemanager.cpp	2013-01-23 00:06:50.677203650 +0100
@@ -55,8 +55,8 @@ LanguageManager::__getInstalledLanguages
   int currentId = 0;
   
   QDir translationsDir(TRANSLATIONS_DIR);
-  QStringList locales = translationsDir.entryList({"vatsinator-*.qm"});
-  QStringList descriptions = translationsDir.entryList({"*.language"});
+  QStringList locales = translationsDir.entryList(QStringList("vatsinator-*.qm"));
+  QStringList descriptions = translationsDir.entryList(QStringList("*.language"));
   
   for (const QString& l: locales) {
     QString locale = l.section('-', 1, 1).section('.', 0, 0);
