diff -apurN a/quazip/ioapi.h b/quazip/ioapi.h
--- a/quazip/ioapi.h	2011-05-26 19:26:16.000000000 +0200
+++ b/quazip/ioapi.h	2013-02-28 01:10:55.035322701 +0100
@@ -36,6 +36,10 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
+  
+#ifndef OF
+# define OF(x) x
+#endif
 
 typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, voidpf file, int mode));
 typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
